Expanded bool class description with examples

Maciej-Gorczyca opened the pull request:
I expanded boolean long description with example of when booleans could be use practice because

Boolean

Description:

Boolean built-in type.

was pretty vague. As new to programming person it took me longer than it should have to understand what those are and why I would want to use them. Hopefully this will make it clearer to new users reading documentation

Feniks-Gaming commented:
@Calinou is that what you meant here https://github.com/godotengine/godot-docs/pull/3059?

Calinou commented:
@Feniks-Gaming Yes :slightly_smiling_face:

YeldhamDev commented:
I think the pronunciation part is kinda unnecessary. And the code should be indented to be right the `[codeblock]` tags.

Feniks-Gaming commented:
I may need some advice here @YeldhamDev this is how code looks in my editor 

![1](https://user-images.githubusercontent.com/30668026/72263758-69776e00-3611-11ea-81bd-e4db408a3e39.PNG)

but when I push it here indentation is all wonky. I am puzzled at to why. Any ideas?

YeldhamDev commented:
It seems that you're indenting the code with spaces instead of tabs. That could be the cause.

Also, add a period at the ending of the comments.

Feniks-Gaming commented:
You was right for some reason there was mix of spaces and tabs it should work now.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+		Boolean is built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like `if` statements.
+
+		For example:
+
```
Remove this space.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+
+		func shoot():
+        	if can_shoot:
+            	# Perform shooting actions here.
```
Add an extra indent here.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+		As you can see above `if can_shoot` is equivalent of `if can_shoot == true`. It is good practice to follow the natural spoken language structure when possible. Use `if can_shoot` rather than `if can_shoot == true` and use `if not can_shoot` rather than `if can_shoot == false`.
+
+		You can use booleans to create more complex logic for example:
+
```
Remove this line.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+		You can use booleans to create more complex logic for example:
+
+		[codeblock]
+		var can_shoot = true # Assigns true values to can_shoot
```
Add a period here.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+		The above code will only create a bullet if both condition are met action "shoot" is pressed and if can_shoot is true. One of practical use cases could be implementing a cool down on player weapon.
+
+		For example above code could be expanded further to:
+
```
Remove this line.

Feniks-Gaming replied:
Don't you think it's a good example for a reader that booleans are used more often and don't have to appear in form of `my_bool = true`?

YeldhamDev replied:
I don't understand your question.

Feniks-Gaming replied:
You said remove this line. I don't know why or am I getting confused and this comment is no longer up to date?

YeldhamDev replied:
The lines that I'm referring to be removed are the empty ones.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+
+		[codeblock]
+		var can_shoot = true # Assigns true values to can_shoot.
+		on ready var cool_down = $CoolDownTimer
```
`onready`, single.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+
+
+			func _on_CoolDownTimer_timeout():
+			can_shoot = true
```
Add an extra indent here.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
 	</brief_description>
 	<description>
-		Boolean built-in type.
+		Boolean is built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like `if` statements.
```
Use `[code][/code]` instead of "`".

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+            	# Perform shooting actions here.
+		[/codeblock]
+
+		As you can see above `if can_shoot` is equivalent of `if can_shoot == true`. It is good practice to follow the natural spoken language structure when possible. Use `if can_shoot` rather than `if can_shoot == true` and use `if not can_shoot` rather than `if can_shoot == false`.
```
Use `[code][/code]` instead of "`".

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,55 @@
+				create_bullet()
+		[/codeblock]
+
+		Please note that `Input.is_action_pressed("shoot")` is also a boolean that is true when "shoot" is pressed and false when `shoot` isn't pressed.
```
Use `[code][/code]` instead of "`".

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,49 @@
+
+
+		func shoot():
+        	if can_shoot:
```
Remember, use tabs.

Maciej-Gorczyca committed:
Expanded class_bool Description with examples

I expanded boolean long description with example of when booleans could be use practice because

Boolean

Description:

Boolean built-in type.

was pretty vague. As new to programming person it took me longer than it should have to understand what those are and why I would want to use them. Hopefully this will make it clearer to new users reading documentation
Update bool.xml

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,49 @@
 	</brief_description>
 	<description>
-		Boolean built-in type.
+		Boolean is built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like [code]if[/code] statements.
```
"Boolean is **a** built-in type".

Feniks-Gaming commented:
Okay all is in place now I just don't understand why some checks are failed if you would be so kind to help with this we could move on @YeldhamDev or @Calinou

YeldhamDev commented:
You didn't need to remove every single empty line, please put back the ones after the `[/codeblock]`.

Also, one mistake on my part, the indents leading to the codeblock need to tabs, but inside needs to be 4 spaces, as in:
```
<tab> [codeblock]
<tab> <4 spaces> if stuff:
```

Feniks-Gaming commented:
Okay all done hope this is now satisfactory

YeldhamDev commented:
OK, last change: I think the descriptions of the codes should be put above rather than below, as in:
```
The code below does stuff:
[codeblock]
```

rather than:
```
[codeblock]
The code above does stuff.
```

Feniks-Gaming commented:
You may be right I changed that. If there is anything else let me know.

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,44 @@
 	<description>
-		Boolean built-in type.
+		Boolean is a built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like [code]if[/code] statements.
+		Please notice that in a code below [code]if can_shoot[/code] is equivalent of [code]if can_shoot == true[/code]. It is good practice to follow the natural spoken language structure when possible. Use [code]if can_shoot[/code] rather than [code]if can_shoot == true[/code] and use [code]if not can_shoot[/code] rather than [code]if can_shoot == false[/code].
```
Just add an empty line here and we're done!

YeldhamDev reviewed (approved):
Happy first commit! :wink:

Feniks-Gaming commented:
Thank you hopefully first of many

YeldhamDev commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		        # Perform shooting actions here.
+		[/codeblock]
+
+		The following code will only create a bullet if both condition are met action "shoot" is pressed and if [code]can_shoot[/code] is true. One of practical use cases could be implementing a cool down on player weapon.
```
OK, one more small issue. I should be here:
"if both condition**s** are met: action blah blah blah"

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
 	<description>
-		Boolean built-in type.
+		Boolean is a built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like [code]if[/code] statements.
+
```
There should not be empty lines in the `<description>` apart from within codeblocks. Each line is a separate paragraph, so this is an empty paragraph (`<p></p>`).

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
 	</brief_description>
 	<description>
-		Boolean built-in type.
+		Boolean is a built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like [code]if[/code] statements.
```
"think of it as **an** on or off (1 or 0) switch"

That's a lot of 2-letter words in a row, might not be easy to read :)

akien-mga replied:
Apart from the first line where "true" and "false" are used as adjectives, all other instances should use code markup with `[code]true[/code]` and `[code]false[/code]` since they refer to keywords.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
-		Boolean built-in type.
+		Boolean is a built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like [code]if[/code] statements.
+
+		Please notice that in a code below [code]if can_shoot[/code] is equivalent of [code]if can_shoot == true[/code]. It is good practice to follow the natural spoken language structure when possible. Use [code]if can_shoot[/code] rather than [code]if can_shoot == true[/code] and use [code]if not can_shoot[/code] rather than [code]if can_shoot == false[/code].
```
I'd move this *after* the first codeblock, and make it a `[b]Note:[/b] In the above code, ...`.

akien-mga replied:
Nevermind, further code blocks are also described before the code, so this would add an inconsistency.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		[codeblock]
+		var can_shoot = true # Assigns true value to can_shoot.
+
+
```
A single separation line before/between functions should be sufficient for such short code snippets.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		        # Perform shooting actions here.
+		[/codeblock]
+
+		The following code will only create a bullet if both conditions are met action "shoot" is pressed and if [code]can_shoot[/code] is true. One of practical use cases could be implementing a cool down on player weapon.
```
"are met<b>:</b>"

akien-mga replied:
I wouldn't mention this specific practical use case, it's very game-specific and I don't think the general `bool` documentation should go too much in this direction.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+
+		Please notice that in a code below [code]if can_shoot[/code] is equivalent of [code]if can_shoot == true[/code]. It is good practice to follow the natural spoken language structure when possible. Use [code]if can_shoot[/code] rather than [code]if can_shoot == true[/code] and use [code]if not can_shoot[/code] rather than [code]if can_shoot == false[/code].
+		[codeblock]
+		var can_shoot = true # Assigns true value to can_shoot.
```
The comment is not really useful. The `bool` class documentation is not the place to document what variable assignment is.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		The following code will only create a bullet if both conditions are met action "shoot" is pressed and if [code]can_shoot[/code] is true. One of practical use cases could be implementing a cool down on player weapon.
+		Please note that [code]Input.is_action_pressed("shoot")[/code] is also a boolean that is true when "shoot" is pressed and false when "shoot" isn't pressed.
+		[codeblock]
+		var can_shoot = true # Assigns true values to can_shoot.
```
Typo in the comment, but as mentioned above I'd suggest removing it.

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		        create_bullet()
+		[/codeblock]
+
+		The following code will set [code]can_shoot[/code] to false and start a timer. This will prevent player from shooting until timer runs out. Next [code]can_shoot[/code] will be set to true again allowing player to shoot once again.
```
"until **the** timer"

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+		    if can_shoot and Input.is_action_pressed("shoot"):
+		        create_bullet()
+		        can_shoot = false
+		        cool_down.start
```
Syntax error :)

akien-mga commented on doc/classes/bool.xml:
```diff
@@ -4,7 +4,45 @@
+
+		The following code will set [code]can_shoot[/code] to false and start a timer. This will prevent player from shooting until timer runs out. Next [code]can_shoot[/code] will be set to true again allowing player to shoot once again.
+		[codeblock]
+		var can_shoot = true # Assigns true values to can_shoot.
```
Same here, typo but unnecessary.

akien-mga reviewed (commented):
Looks good overall, left some comments for final style adjustments to make.

Feniks-Gaming commented:
@akien-mga I believe I added all the changes you wanted if there is anything else that needs doing let me know.

akien-mga commented:
Thanks for your contribution and for putting up with our reviews :)

Feniks-Gaming commented:
Thanks for reviewing I actually enjoyed it :)
